src/Compilers/CSharp/Test/Emit2/Semantics/InlineArrayTests.cs

```diff
@@ -7439,6 +7439,74 @@ .maxstack  2
 ");
         }
 
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void ElementAccess_Value_07()
+        {
+            var src = @"
+class Program
+{
+    static System.Span<int> M1()
+    {
+        System.Span<int> x = stackalloc int[2];
+        return M3(x)[0];
+    }
+
+    static System.Span<int> M2()
+    {
+        System.Span<int> x = stackalloc int[2];
+        var y = M3(x)[0];
+        return y;
+    }
+
+    static Buffer10 M3(System.Span<int> x)
+    {
+        throw null;
+    }
+
+    static System.Span<int> M1(System.Span<int> xx)
+    {
+        return M3(xx)[0];
+    }
+
+    static System.Span<int> M2(System.Span<int> xx)
+    {
+        var yy = M3(xx)[0];
+        return yy;
+    }
+}
+
+[System.Runtime.CompilerServices.InlineArray(10)]
+public ref struct Buffer10
+{
+    private System.Span<int> _element0;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll);
+            comp.VerifyDiagnostics(
+                // (7,16): error CS0306: The type 'Span<int>' may not be used as a type argument
```

Comment:
Not understanding this error: where is the type argument happening here.

Reply:
> Not understanding this error: where is the type argument happening here. 

In general, an access to inline array elements is performed via a Span. In order to do this, the element type must be a valid type argument.

Reply:
Chatted offline, I understand where the error is coming from now. It's the synthensized indexers cause the element type is `Span<T>`